flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/serializer/AvroToProtoSerializer.java

```diff
@@ -191,7 +191,7 @@ private static Object toProtoValue(
                 if (convertedValue != value) {
                     return convertedValue;
                 }
-                return Long.valueOf(value.toString());
+                return ((Number) value).longValue();
```

Comment:
This can be (It does the same AFAIK):
```suggestion
                return (long) value;
```